Can't compile PPSSPP on the Raspberry Pi 4

### What happens?
When I compile with:
`cmake . -DCMAKE_TOOLCHAIN_FILE=cmake/Toolchains/raspberry.armv7.cmake`
it fails with this error when making:
```
[ 51%] Building CXX object CMakeFiles/Core.dir/GPU/Common/TextureDecoderNEON.cpp.o
/tmp/ccJDi8IY.s: Assembler messages:
/tmp/ccJDi8IY.s:37: Error: selected processor does not support `movw r0,0xc00b' in ARM mode
/tmp/ccJDi8IY.s:38: Error: selected processor does not support `movt r0,0x9bd9' in ARM mode
/tmp/ccJDi8IY.s:39: Error: selected processor does not support `movw r1,0x4b73' in ARM mode
/tmp/ccJDi8IY.s:40: Error: selected processor does not support `movt r1,0xb651' in ARM mode
/tmp/ccJDi8IY.s:42: Error: selected processor does not support `movw r0,0x4d9b' in ARM mode
/tmp/ccJDi8IY.s:43: Error: selected processor does not support `movt r0,0x4309' in ARM mode
/tmp/ccJDi8IY.s:44: Error: selected processor does not support `movw r1,0x0083' in ARM mode
/tmp/ccJDi8IY.s:45: Error: selected processor does not support `movt r1,0x0001' in ARM mode
/tmp/ccJDi8IY.s:47: Error: selected processor does not support `movw r0,0x2455' in ARM mode
make[2]: *** [CMakeFiles/Core.dir/build.make:3196: CMakeFiles/Core.dir/GPU/Common/TextureDecoderNEON.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:496: CMakeFiles/Core.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
```

I've also made a new raspberry.armv8.cmake file in which I changed the mcpu to cortex-a72.
Gives me the same error. 

Is this a bug? Or am I doing something wrong?

--- a/cmake/Toolchains/raspberry.armv7.cmake
+++ b/cmake/Toolchains/raspberry.armv7.cmake
@@ -1,4 +1,10 @@
-if(NOT EXISTS "/opt/vc/include/bcm_host.h")
+file(READ "/sys/firmware/devicetree/base/compatible" PPSSPP_PI_MODEL)
+
+if(PPSSPP_PI_MODEL MATCHES "raspberrypi,4")
+  set(PPSSPP_PI_MODEL4 ON)
+endif()
+
+if(NOT PPSSPP_PI_MODEL4 AND NOT EXISTS "/opt/vc/include/bcm_host.h")
   message(FATAL_ERROR "RaspberryPI platform not recognized")
 endif()
 
@@ -8,21 +14,29 @@ include_directories(SYSTEM
   /opt/vc/include/interface/vmcx_host/linux
 )
 
-add_definitions(
-  -DPPSSPP_PLATFORM_RPI=1
-  -U__GCC_HAVE_SYNC_COMPARE_AND_SWAP_2
-)
+if(NOT PPSSPP_PI_MODEL4)
+  add_definitions(
+    -DPPSSPP_PLATFORM_RPI=1
+    -U__GCC_HAVE_SYNC_COMPARE_AND_SWAP_2
+  )
+endif()
 
-set(ARCH_FLAGS "-mcpu=cortex-a7 -mfpu=neon -mfloat-abi=hard")
+if(PPSSPP_PI_MODEL4)
+  set(ARCH_FLAGS "-march=armv7-a -mcpu=cortex-a72 -mtune=cortex-a72 -mfpu=neon-vfpv4 -mfloat-abi=hard")
+else()
+  set(ARCH_FLAGS "-mcpu=cortex-a7 -mfpu=neon -mfloat-abi=hard")
+endif()
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${ARCH_FLAGS}"  CACHE STRING "" FORCE)
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${ARCH_FLAGS}" CACHE STRING "" FORCE)
 set(CMAKE_ASM_FLAGS "${CMAKE_ASM_FLAGS} ${ARCH_FLAGS}" CACHE STRING "" FORCE)
 
 set(CMAKE_EXE_LINKER_FLAGS "-L/opt/vc/lib" CACHE STRING "" FORCE)
 
-set(OPENGL_LIBRARIES brcmGLESv2 bcm_host)
-set(EGL_LIBRARIES brcmEGL)
+if(NOT PPSSPP_PI_MODEL4)
+  set(OPENGL_LIBRARIES brcmGLESv2 bcm_host)
+  set(EGL_LIBRARIES brcmEGL)
+  set(USING_FBDEV ON)
+endif()
 set(USING_GLES2 ON)
-set(USING_FBDEV ON)
 set(ARMV7 ON)
 set(USE_WAYLAND_WSI OFF)

